DRODLib/Stalwart.cpp

```diff
@@ -217,7 +217,7 @@ bool CStalwart::GetGoal(UINT& wX, UINT& wY)
 	const CMonster *pMonster = this->pCurrentGame->pRoom->pFirstMonster;
 	while (pMonster)
 	{
-		if (CStalwart::typesToAttack.has(pMonster->GetIdentity()))
+		if (CStalwart::typesToAttack.has(pMonster->wType) && pMonster->IsPlayerAllyTarget())
```

Comment:
Pardon, don't have the project open in front of me.  Am I correct in understanding this change will not alter pre-existing logic of how stalwarts will target characters with a certain appearance?

That is, previously, a stalwart won't attack an NPC of type Roach, and now it still won't, unless the Ally Target flag is set, correct?

Reply:
This is correct. NPCs can now be considered for targeting, but the behavior flag is required for them to actually "become" a target. Otherwise, Stalwarts and Soldiers remain as blind to NPCs as they've always been.